src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs

```diff
@@ -123,6 +147,30 @@ internal static ObsoleteDiagnosticKind GetObsoleteDiagnosticKind(Symbol symbol,
                     // later stage
                     return ObsoleteDiagnosticKind.LazyPotentiallySuppressed;
             }
+
+            static bool isExperimentalSymbol(NamedTypeSymbol namedTypeSymbol)
+            {
+                return namedTypeSymbol.Arity == 0
+                    && namedTypeSymbol.HasNameQualifier("System.Diagnostics.CodeAnalysis")
+                    && namedTypeSymbol.Name.Equals("ExperimentalAttribute", StringComparison.Ordinal);
+            }
+
+            static ObsoleteDiagnosticKind getExperimentalDiagnosticKind(Symbol containingMember, bool forceComplete)
+            {
+                switch (GetObsoleteContextState(containingMember, forceComplete, getStateFromSymbol: static (symbol) => symbol.ExperimentalState))
```

Comment:
>switch [](http://example.com/codeflow?start=16&length=6)

It looks like the only difference between this `switch` and the one above is the lambda. Consider refactoring to remove the local function and share the `switch (GetObsoleteContextState(...)) { }` statement.
```
Func<Symbol, ThreeState> getStateFromSymbol;

switch (symbol.ObsoleteKind)
{
    ...
}

switch (GetObsoleteContextState(containingMember, forceComplete, getStateFromSymbol)
{
    ...
}
```